internal/fetchclient/fetchclient.go

```diff
@@ -196,12 +207,36 @@ func (c *Client) fetchMaven(ctx context.Context, group string, name string) (str
 		} `json:"response"`
 	}
 	if err := json.NewDecoder(response.Body).Decode(&data); err != nil {
-		return "", err
+		return nil, err
 	}
+	// TODO: support pagination over additional results using Response.NumFound / Response.Start
 	if len(data.Response.Docs) == 0 {
-		return "", errors.New("failed to get versions from response docs")
+		return nil, errors.New("failed to get versions from response docs")
+	}
+	results := &mavenResults{
+		latestPatchVersions: make(map[string]string),
+	}
+	for _, doc := range data.Response.Docs {
+		version := doc.Version
+		if !strings.HasPrefix(version, "v") {
+			version = "v" + version
+		}
+		if !semver.IsValid(version) || semver.Prerelease(version) != "" {
+			continue
+		}
+		version = semver.Canonical(version)
+		if results.latestVersion == "" || semver.Compare(results.latestVersion, version) < 0 {
+			results.latestVersion = version
+		}
+		release := semver.MajorMinor(version)
+		if results.latestPatchVersions[release] == "" || semver.Compare(results.latestPatchVersions[release], version) < 0 {
+			results.latestPatchVersions[release] = version
```

Comment:
Keep track of the latest version for a given major + minor release. We could use this in a follow up PR to update to the latest patch version of a plugin we're already shipping.